### PR TITLE
[New Extension] Next Run

### PR DESCRIPTION
**Pull request by jonathangiardino**
## Description

Spin up a Next.js project from Raycast

## Screencast

next-run-1
next-run-2
next-run-3
next-run-4
next-run-5
next-run-6

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/next-run/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    getSavedPreferences();
+  }, []);
+
+  return (
+    <Form
+      navigationTitle={`Clone from "${item.name}"`}
+      actions={
+        <ActionPanel>
+          {location === "local" ? (
+            <Action.SubmitForm icon={Icon.Code} title="Clone locally" onSubmit={async () => await cloneLocally()} />
```
Comment (pernielsentikaer) [inline extensions/next-run/src/index.tsx:127]:
```suggestion
            <Action.SubmitForm icon={Icon.Code} title="Clone Locally" onSubmit={async () => await cloneLocally()} />
```

File: extensions/next-run/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        info="Clone the project locally or deploy it directly to Vercel."
+      >
+        <Form.Dropdown.Item value="local" title="Local" icon={Icon.Terminal} />
+        <Form.Dropdown.Item value="remote" title="Vercel" icon={Icon.Globe} />
+      </Form.Dropdown>
+      {location === "local" ? (
+        <>
+          <Form.FilePicker
+            id="folder"
+            title="Preferred folder"
```
Comment (pernielsentikaer) [inline extensions/next-run/src/index.tsx:161]:
```suggestion
            title="Preferred Folder"
```

File: extensions/next-run/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+              url={vercelUrl}
+              onOpen={async () => cloneOnVercel()}
+            />
+          )}
+        </ActionPanel>
+      }
+    >
+      <Form.TextField
+        id="projectName"
+        title="Project name"
```
Comment (pernielsentikaer) [inline extensions/next-run/src/index.tsx:141]:
```suggestion
        title="Project Name"
```

File: extensions/next-run/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    <List isLoading={isLoading} searchBarPlaceholder="Search Examples">
+      {Array.isArray(data) && data?.length
+        ? data?.map((item: ExampleSource) => (
+            <List.Item
+              key={item.name}
+              icon={Icon.Folder}
+              title={item.name}
+              actions={
+                <ActionPanel>
+                  <Action.Push title="Project Setup" target={<ProjectForm item={item} />} />
```
Comment (pernielsentikaer) [inline extensions/next-run/src/index.tsx:54]:
```suggestion
                  <Action.Push title="Project Setup" icon={Icon.Hammer} target={<ProjectForm item={item} />} />
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

* It's possible to submit the form without a name; you should add [Validation](https://developers.raycast.com/api-reference/user-interface/form#validation) or use [useForm](https://developers.raycast.com/utilities/react-hooks/useform)

* Maybe add a hammer, wand, or something else like this to make it look even crisper

Raycast-z9mMRD75@2x

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by jonathangiardino** [comment]
Hey, @pernielsentikaer I have applied all the changes! 🚀

**Review by pernielsentikaer — APPROVED** [review]
Hello 👋

I gave it a quick spin, works as expected, approved 🔥

**Pull request merged**